packages/auth/src/grant/model.ts

```diff
@@ -147,3 +154,17 @@ export function isPendingGrant(grant: Grant): grant is PendingGrant {
     grant.interactNonce
   )
 }
+
+export function isRejectedGrant(grant: Grant): boolean {
+  return !!(
+    grant.state === GrantState.Finalized &&
+    grant.finalizationReason === GrantFinalization.Rejected
```

Comment:
Can there be a scenario where `finalizationReason` is `Rejected` but `state !== Finalized`?

Reply:
I'm just wondering why we are checking both but I guess it's a sanity check of our data, too.